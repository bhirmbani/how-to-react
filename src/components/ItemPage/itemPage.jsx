// @flow

import React from "react";

// import mockData from "../../mock/api.json";
import ItemDetails from "../ItemDetails/itemDetails";
import filterItem from "../modules/filterItem";

type State = {
  searchInput: string,
}

class ItemPage extends React.Component<{}, State> {
  state = {
    searchInput: ''
  };

  handleSearchInput = (e: SyntheticKeyboardEvent<HTMLInputElement>) => {
    this.setState({ searchInput: e.currentTarget.value });
  };

  render() {
    return (
      <div>
        <input
          onChange={this.handleSearchInput}
          value={this.state.searchInput}
          type="text"
          placeholder="Search..."
        />
        {filterItem(this.state.searchInput).map(item => (
          <ItemDetails key={item.id} item={item} />
        ))}
      </div>
    );
  }
}

export default ItemPage;
