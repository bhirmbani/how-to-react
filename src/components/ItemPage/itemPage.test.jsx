import React from "react";
import renderer from "react-test-renderer";
import ItemPage from "./itemPage";

test("ItemPage renders correctly", () => {
  const component = renderer.create(<ItemPage />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
