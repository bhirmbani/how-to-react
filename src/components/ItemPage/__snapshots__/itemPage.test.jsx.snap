// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemPage renders correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
  <div
    className="sc-bdVaJa icLNKT"
  >
    <h3>
      TV
    </h3>
    <p>
      2500000
    </p>
  </div>
  <div
    className="sc-bdVaJa icLNKT"
  >
    <h3>
      Hape
    </h3>
    <p>
      9700000
    </p>
  </div>
  <div
    className="sc-bdVaJa icLNKT"
  >
    <h3>
      Celana Chino
    </h3>
    <p>
      250000
    </p>
  </div>
</div>
`;
